spec/requests/registrations_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+
+RSpec.describe 'Registrations', type: :request do
+  let(:user) { create :user }
+  let(:valid_attributes) { attributes_for :user }
+
+  describe 'POST /create' do
+    it 'creates a new User' do
+      expect do
+        post '/api/auth', params: valid_attributes
+      end.to change(User, :count).by(1)
+    end
+  end
+
+  describe 'PUT /update' do
+    let(:new_name) { Faker::Name.name }
+
+    it 'updates the user' do
+      put '/api/auth',
+          params: { name: new_name }, headers: user.create_new_auth_token, as: :json
+      user.reload
```

Comment:
Third comment about the same thing, but just for some context if you set this request to run before each test case, you don't need to reload the user here, since it'll be already updated because the request for the test was made before entering the it block.